### ISSUE TITLE
Status code in response does not match the status_code in the decorator

### ISSUE THREAD

**Issue by adityaguru149** [tiangolo/fastapi#2097]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug. - If this is a bug, I can try raising a PR

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from typing import Optional
from fastapi import FastAPI
from fastapi.responses import JSONResponse
from pydantic import BaseModel
from starlette import status


class MyItem(BaseModel):
    item_id: str
    is_offer: Optional[bool] = None


app = FastAPI()


@app.post("/items", status_code=status.HTTP_201_CREATED)
async def post_item(item: MyItem):
    return JSONResponse({"item_id": item.item_id})

```

### Description
New user of FastAPI here. I have considerable experience with flask where return codes are set in the return statement.
Took an [example](https://fastapi.tiangolo.com/tutorial/response-status-code/) from the documentation and gave it a run to test my understanding.
Swagger Docs respond correctly.

* POST a request to the endpoint `/items` (default example in swagger).
* It returns with status code `200`.
* But I expected it to return `201`.

Is this expected? Have I misunderstood something in the docs?
My understanding of the status_code parameter is that it is the default status code which can be overridden by `<Response>.status_code = <int>`
Is `status_code` just for the Swagger docs?

### Environment

* OS: [e.g. Linux / Windows / macOS]: Ubuntu
* FastAPI Version [e.g. 0.3.0]: 0.61.1 (even tried with 0.60.0 to check if it is just happening for this version)
* Python Version: 3.7.5

### Additional context
![image](https://user-images.githubusercontent.com/17622156/94315337-f09e2380-ff9f-11ea-853e-8af6df3911b8.png)

**Comment by gabrielweich** [comment]
If you return only a dictionary it will respect the router's status_code, but when returning a JSONResponse it will use the status_code passed to JSONResponse (which by default is 200).

**Comment by adityaguru149** [comment]
Thanks a lot @gabrielweich. That was super fast and spot on.

**Q to the maintainers:**
Is this expected behavior? For some reason was this designed to be this way?

I think it would be more consistent and easier to reason with if the status_code in decorator is respected with precedence order:
explicitly status_code in JSONResponse/Response > status_code set in the decorator > the default 200 status_code (that is anyhow the default for the decorator too)

It might just be my personal preference, so, please feel free to let me know your thoughts and close the issue.

**Comment by Kludex** [comment]
> Is this expected behavior? For some reason was this designed to be this way?
Yes, it's the expected behavior. It was a choice, it could have been different.

When you return `JSONResponse` you're basically saying: "Hey, I want to be in charge now!". Then FastAPI loses some of its power.

**Comment by tiangolo** [comment]
Yep, exactly what @Kludex said. 👆
